assimilation_code/modules/assimilation/assim_tools_mod.pf.f90

```diff
@@ -1244,18 +1497,23 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
       endif
 
       ! Get the initial values of inflation for this variable if state varying inflation
-      if(local_varying_ss_inflate) then
+      if(local_varying_ss_inflate .and. filter_kind /= 9) then
          varying_ss_inflate    = ens_handle%copies(ENS_INF_COPY,    state_index)
          varying_ss_inflate_sd = ens_handle%copies(ENS_INF_SD_COPY, state_index)
       else
          varying_ss_inflate    = 0.0_r8
          varying_ss_inflate_sd = 0.0_r8
       endif
-
-      ! Compute the distance and covariance factor
+     
+      ! Compute the distance and covariance factor 
       cov_factor = comp_cov_factor(close_state_dist(j), cutoff_rev, &
          base_obs_loc, base_obs_type, my_state_loc(state_index), my_state_kind(state_index))
 
+      ! SEEMS TO BE NEEDED FOR VERTICAL LOCALIZATION IN BGRID
+      if (filter_kind == 9) then
+         if ( cov_factor == 0.0_r8 ) cycle STATE_UPDATE
```

Comment:
this code has moved and is now inside a PF-filter only if-test but all filters need to cycle if the covariance value is 0.  also the test should be after the impact_factor multiplication.

Reply:
since this version is only going to be run with filter_kind == 9, this could stay as-is but maybe a comment should be added that other filter kinds will also need to cycle.  this test should still move down below the impact_factor multiply code so the impact code can be tested and used with this filter_kind and work correctly.